src/shared/gpt.c

```diff
@@ -158,6 +158,23 @@ static const char *const partition_designator_table[] = {
 
 DEFINE_STRING_TABLE_LOOKUP(partition_designator, PartitionDesignator);
 
+static const char *const partition_mountpoint_table[] = {
+        [PARTITION_ROOT]                      = "/",
+        [PARTITION_ROOT_SECONDARY]            = "/",
+        [PARTITION_ROOT_OTHER]                = "/",
+        [PARTITION_USR]                       = "/usr",
+        [PARTITION_USR_SECONDARY]             = "/usr",
+        [PARTITION_USR_OTHER]                 = "/usr",
+        [PARTITION_HOME]                      = "/home",
+        [PARTITION_SRV]                       = "/srv",
+        [PARTITION_ESP]                       = "/efi",
```

Comment:
This could be /boot/efi in some cases right? Could this cause issues?

Reply:
it is not a symlink, on debian/ubuntu it's the mountpoint where the ESP is mounted on

```
$ ls -l /boot | grep efi
drwx------ 4 root root     4096 Jan  1  1970 efi
```

Reply:
The discoverable partitions spec mentions that the EFI system partition is either mounted to /efi or /boot as a fallback. So I don't think we should support /boot/efi unless we add it to the spec first. That still leaves /boot however. We can turn the table values into a nulstr of mountpoints so we can support more than one mount point for the ESP partition. For the ESP, we'll ignore both /boot and /efi.

Reply:
yeah that's fine, the important thing is to avoid regressions

Reply:
Implemented the nulstr approach